OpenRA.Game/GameRules/Ruleset.cs

```diff
@@ -87,16 +87,19 @@ public class Ruleset
 
 		public IEnumerable<KeyValuePair<string, MusicInfo>> InstalledMusic { get { return Music.Where(m => m.Value.Exists); } }
 
-		static IReadOnlyDictionary<string, T> MergeOrDefault<T>(string name, IReadOnlyFileSystem fileSystem, IEnumerable<string> files, MiniYaml additional,
-			IReadOnlyDictionary<string, T> defaults, Func<MiniYamlNode, T> makeObject)
+		static IReadOnlyDictionary<string, T> MergeOrDefault<T>(string name,
```

Comment:
IMO a cleaner approach here would be to pass in a `Func<MiniYamlNode, bool> filterNode` argument, then have the callers pass in either `n => .StartsWith("^", StringComparison.Ordinal)` (for actors) or `_ => true` (everything else).

Reply:
Makes sense. I went with a default behavior of no-op rather than requiring explicit behavior on the other callers.